opentelemetry-api/src/opentelemetry/context/__init__.py

```diff
@@ -138,15 +138,119 @@ def __repr__(self):
         asyncio.run(main())
 """
 
+import typing
+
+from contextlib import contextmanager
+
 from .base_context import BaseRuntimeContext
 
 __all__ = ["Context"]
 
 try:
     from .async_context import AsyncRuntimeContext
 
-    Context = AsyncRuntimeContext()  # type: BaseRuntimeContext
+    _CONTEXT = AsyncRuntimeContext()  # type: BaseRuntimeContext
 except ImportError:
     from .thread_local_context import ThreadLocalRuntimeContext
 
-    Context = ThreadLocalRuntimeContext()
+    _CONTEXT = ThreadLocalRuntimeContext()
+
+
+class Context:
```

Comment:
This may be one too many layers of context API onion.

As I understand it, the point of adding this module in https://github.com/open-telemetry/opentelemetry-python/pull/57 was largely the same as in this PR: to provide a single global context, and hide access to the context behind an API so we could swap out the implementation as needed.

If we didn't have to worry about the `contextvars` package missing (or exotic 3p threading models, about which more later), this could have reasonably been written without a`Context` object at all. In that case `opentelemetry.context` would provide two methods:
- `get_current() -> ContextVar[Dict[str, object]]`
- `set_current(context: ContextVar[Dict[str, object]]) -> None`

or, hiding the contextvar:
- `get_current() -> Dict[str, object]`
- `set_current(context: Dict[str, object]) -> None`

As compared to this PR, what would we lose with this simpler approach?

What would we lose if `AsyncRuntimeContext` and `ThreadLocalRuntimeContext` implemented `Context`, instead of `Context` being composed of one of these?